utils/navigator_gui/navigator_gui/shooter.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+
+'''
+Shooter: A graphical interface to view the status of and control the shooter
+mechanism.
+'''
+
+
+from __future__ import division
+
+import os
+
+from navigator_alarm import AlarmListener
+from python_qt_binding import QtGui
+from python_qt_binding import loadUi
+from qt_gui.plugin import Plugin
+from remote_control_lib import RemoteControl
+import rospkg
+import rospy
+from std_msgs.msg import String
+
+
+__author__ = "Anthony Olive"
+__maintainer__ = "Anthony Olive"
+__email__ = "anthony@iris-systems.net"
+__copyright__ = "Copyright 2016, MIL"
+__license__ = "MIT"
+
+
+class Shooter(Plugin):
+
+    def __init__(self, context):
+        super(Shooter, self).__init__(context)
+
+        # Create the widget and name it
+        self._widget = QtGui.QWidget()
+        self._widget.setObjectName("Shooter")
+        self.setObjectName("Shooter")
+
+        # Extend the widget with all attributes and children in the UI file
+        ui_file = os.path.join(rospkg.RosPack().get_path("navigator_gui"), "resource", "shooter.ui")
+        loadUi(ui_file, self._widget)
+
+        self.is_killed = False
+        self.remote = RemoteControl("shooter gui")
+        self.remote.is_timed_out = True
+
+        self.shooter_status = {
+            "received": "Unknown",
+            "cached": "Unknown"
+        }
+        self.disc_speed_setting = 0
+
+        self.connect_ui()
+
+        self.kill_listener = AlarmListener("kill", self.update_kill_status)
```

Comment:
This is not implemented in the remote control library. None of the functions there are dependent on the killed state because they assume the user is not trying to do something stupid (e.g. there may be a scenario where we want to be able to test the shooter while killed). If you think this is a good behavior to implement, I think it would be best to leave it on the actual services that run on Navigator.